### PR TITLE
Reject promise of hook throws an error

### PR DESCRIPTION
**Pull request by just-boris**
Before this change if asynchronous hook immediately throws an error, it is not considered as failed.

Actually in the test only `hookReject` made sense, the result `hookThrows` was ignored, I fixed test, so now both use cases are properly tested.

### CODE REVIEW

**Comment by just-boris** [comment]
@christian-bromann ping.
This fix is important to pass tests in wdio-allure-reporter

File: index.js [hunk 1]
```diff
@@ -96,6 +96,7 @@ let executeHooksWithArgs = (hooks = [], args) => {
                 commandIsRunning = _commandIsRunning
             }).run()
         } catch (e) {
+            reject(e)
```
Comment (christian-bromann) [inline index.js]: I am not sure if this is a good idea, I don't want that any services can blow up the test suite if they don't work. That is way I went without calling reject.

Comment (just-boris) [inline index.js]: consider the following test suite:

```js
describe('suite', () => {
    before(() => {
        throw new Error('stop test right now!')
    })
    
    it('is a test which should not run', () => {})
})
```

Vanilla Mocha gets the error from `before` and stops the suite. With wdio-sync wrapper around the error will be swallowed and Mocha will run test case like `before` hook went fine.

Do you really think that this is okay to change default behavior of Mocha tests?

Comment (christian-bromann) [inline index.js]: Then we only need to reject the error if it comes from Mocha. It should not affect the wdio hooks.

**Comment by just-boris** [comment]
@christian-bromann okay, now I got the difference between Wdio and Mocha hooks.
So, not so much changes are necessary here, but at least you have got more tests on it.

**Comment by christian-bromann** [comment]
@just-boris thanks, will this fix the wdio-allure-reporter-tests?

**Pull request merged**